### PR TITLE
Fix prerequisites Alien::Base::ModuleBuild and Test::LoadAllModules

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello @Getty 

Thank you for this Alien module.

Here is a proposed change related to #1 

@plicease could you please review

I tested it locally see below : 

`sudo cpanm Alien::mpg123`

```
--> Working on Alien::mpg123
Fetching http://www.cpan.org/authors/id/G/GE/GETTY/Alien-mpg123-0.002.tar.gz ... OK
Configuring Alien-mpg123-0.002 ... N/A
! Configure failed for Alien-mpg123-0.002. See /home/tduponchelle/.cpanm/work/1588154494.31330/build.log for details.
```

After the fix : 
`dzil build`

produces : 
```
[DZ] beginning to build Alien-mpg123
[@Author::GETTY/Git::NextVersion] Bumping version from 0.002 to 0.003
[DZ] guessing dist's main_module is lib/Alien/mpg123.pm
[@Author::GETTY/@Filter/ExtraTests] rewriting author test xt/author/pod-syntax.t
[DZ] writing Alien-mpg123 in Alien-mpg123-0.003
[DZ] building archive with Archive::Tar; install Archive::Tar::Wrapper 0.15 or newer for improved speed
[DZ] writing archive to Alien-mpg123-0.003.tar.gz
[DZ] built in Alien-mpg123-0.003
```

And install is OK
`sudo cpanm Alien-mpg123-0.003.tar.gz`

produces : 

```
--> Working on Alien-mpg123-0.003.tar.gz
Fetching file:///home/tduponchelle/Code/p5-alien-mpg123/Alien-mpg123-0.003.tar.gz ... OK
==> Found dependencies: Alien::Base::ModuleBuild, Test::LoadAllModules
--> Working on Alien::Base::ModuleBuild
Fetching http://www.cpan.org/authors/id/P/PL/PLICEASE/Alien-Base-ModuleBuild-1.14.tar.gz ... OK
Configuring Alien-Base-ModuleBuild-1.14 ... OK
Building and testing Alien-Base-ModuleBuild-1.14 ... OK
Successfully installed Alien-Base-ModuleBuild-1.14
--> Working on Test::LoadAllModules
Fetching http://www.cpan.org/authors/id/K/KI/KITANO/Test-LoadAllModules-0.022.tar.gz ... OK
Configuring Test-LoadAllModules-0.022 ... OK
Building and testing Test-LoadAllModules-0.022 ... OK
Successfully installed Test-LoadAllModules-0.022
Configuring Alien-mpg123-0.003 ... OK
Building and testing Alien-mpg123-0.003 ... OK
Successfully installed Alien-mpg123-0.003
3 distributions installed
```

Best regards.

Thibault

### CODE REVIEW

File: dist.ini [hunk 1]
```diff
@@ -9,3 +9,14 @@ no_makemaker = 1
...
+
+[Prereqs / ConfigureRequires]
+Alien::Base = 0.021
+Alien::Base::ModuleBuild = 0.023
+Test::LoadAllModules = 0
+
+[Prereqs / Requires]
+Alien::Base = 0.021
+Alien::Base::ModuleBuild = 0.023
+Test::LoadAllModules = 0
```
Comment (plicease) [inline dist.ini]: Looks like dzil already pulls in `Test::LoadAllModules`
https://metacpan.org/source/GETTY/Alien-mpg123-0.002/META.json#L38

File: dist.ini [hunk 2]
```diff
@@ -9,3 +9,14 @@ no_makemaker = 1
 alien_repo = http://www.mpg123.de/download
 alien_pattern_suffix = \.tar\.bz2
 alien_bins = mpg123
+
+[Prereqs / ConfigureRequires]
+Alien::Base = 0.021
```
Comment (plicease) [inline dist.ini]: This is okay, but AB isn't actually needed at configure time.

File: dist.ini [hunk 3]
```diff
@@ -9,3 +9,14 @@ no_makemaker = 1
...
+Alien::Base::ModuleBuild = 0.023
+Test::LoadAllModules = 0
+
+[Prereqs / Requires]
+Alien::Base = 0.021
+Alien::Base::ModuleBuild = 0.023
```
Comment (plicease) [inline dist.ini]: ABMB isn't a runtime prereq, so it should only be in configure requires.

**Comment by plicease** [comment]
I would narrow the prereqs as mentioned in the comments 👆 , but in principal this LGTM, and probably the simplest fix.

@Getty could also bump the prereq to a recent `[Alien]` for his bundle and explicitly set the prereq `dist.ini` to the new version of his bundle, since recent versions of the `[Alien]` plugin do this automatically.

File: dist.ini (hunk 1 above)
Comment (thibaultduponchelle) [reply dist.ini]: It is in CPAN, but my tests with latest Dist::Zilla do not put it in META files. Maybe the cpan tarball is not generated with the exact same dist.ini than this github repo ?
I updated to move Test::LoadAllModules in a "[Prereqs / TestRequires]"  section

File: dist.ini (hunk 3 above)
Comment (thibaultduponchelle) [reply dist.ini]: Fixed with new commit, thank you ! :smile:

File: dist.ini (hunk 2 above)
Comment (thibaultduponchelle) [reply dist.ini]: Fixed with new commit. Thank you ! :+1:

**Comment by Getty** [comment]
New version uploaded, added up Test::LoadAllModules requirement, the other one was automatically fixed through the [Alien] of the bundle.

**Comment by thibaultduponchelle** [comment]
Great! :+1:

**Pull request closed without merging**